domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -3,11 +3,109 @@ package dagtraversalmanager_test
 import (
 	"github.com/kaspanet/kaspad/domain/consensus"
 	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/model/testapi"
 	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
 	"github.com/kaspanet/kaspad/domain/dagconfig"
 	"testing"
 )
 
+func TestBlockAtDepth(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, tearDown, err := factory.NewTestConsensus(params, false,
+			"TestBlockAtDepth")
+		if err != nil {
+			t.Fatalf("NewTestConsensus: %s", err)
+		}
+		defer tearDown(false)
+
+		// "checkExpectedBlockChainDag" - compare BlockAtDepth to a loop on the SelectedChainIterator on a single chain DAG.
+		checkExpectedBlockChainDag := func(highHash *externalapi.DomainHash, depth uint64) {
+			currentBlock := highHash
+			currentBlockData, err := tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+			if err != nil {
+				t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
```

Comment:
I know other devs don't do this, but I love it when the t.Fatalf error message has something meaningful inside it e.g. "Failed getting GhostdagData for block with hash %s: %+v"

Reply:
Changed, thanks.